tests/unit/test_platform.cpp

```diff
@@ -967,4 +967,5 @@ OT_TOOL_WEAK void otPlatDnssdStopIp4AddressResolver(otInstance *aInstance, const
 OT_TOOL_WEAK otError otPlatLogCrashDump(void) { return OT_ERROR_NONE; }
 #endif
 
+void otPlatAssertFail(const char *, int) {}
```

Comment:
Let's declare it here as `OT_TOOL_WEAK` so specific tests can override it and provide their own if they want.